pinc/maybe_mail.inc

```diff
@@ -4,6 +4,135 @@ include_once($relPath.'misc.inc'); // html_safe()
+    //          pip install fakeemail
+    //          fakeemail 2025 8080 0.0.0.0
+    //          Open http://mydomain.org:8080/ in browser to monitor emails
+    $mail_mode = 0;
+
+    if ($testing) {
+        echo "\n<hr>\n";
+        echo "\$testing is $testing. If it were FALSE, the following Markdown mail would have been sent:\n";
+        echo "<pre style='white-space: pre-wrap'>\n";
+        echo html_safe("To: $to\n");
+        echo html_safe("Subject: $subject\n");
+        echo html_safe("From: $from\n");
+        echo html_safe("Reply-To: $reply_to\n");
+        echo "</pre>";
+        echo "<pre style='white-space: pre-wrap'>";
+        echo html_safe("$message\n");
+        echo "</pre>\n";
+        echo "<hr>\n";
+        echo "$html_message\n"   ;
+        echo "<hr>\n";
+        // Uncomment below to test emails using localhost:2025
+        $mail_mode = -1;
+    } else {
+        $mail_mode = 1;
+    }
+
+    if ($mail_mode === 0) {
+        return success;    // Nothing left to do!
+    }
+
+    $mail = new PHPMailer(true);
+    try {
+        // Server settings
+        $mail->isSMTP();
+        if ($mail_mode < 0) {   // Testing email using localhost:2025
+            // $mail->SMTPDebug = SMTP::DEBUG_SERVER;        // Uncomment to enable verbose debug output
+            $mail->Host = 'localhost';
+            $mail->Port = 2025;
+            $mail->SMTPAuth = false;
+        } else {    // TODO - Normal production settings
+            // $mail->Host = 'localhost';
+            // $mail->Port = 2025;
+            // $mail->SMTPAuth = false;
+            // $mail->Username   = 'user@example.com';
+            // $mail->Password   = 'secret';
+            // $mail->SMTPSecure = PHPMailer::ENCRYPTION_SMTPS;
+        }
+
+        // Recipients
+        [$name, $address] = get_name_address($from);
+        $mail->setFrom($address, $name);
+        [$name, $address] = get_name_address($reply_to);
+        $mail->addReplyTo($address, $name);
+        $mail->addAddress($to);
+
+        // Content
+        $mail->isHTML(true);
+        $mail->Subject = $subject;
+        $mail->Body = $html_message;
+        $mail->AltBody = $message;
+
+        $mail->send();
+        if ($testing and $mail_mode < 0) {
+            echo "<p>Message sent to localhost</p>\n";
+            echo "<hr>\n";
+        }
+    } catch (Exception $e) {
+        echo "<p>Message could not be sent. Mailer Error: {$mail->ErrorInfo}</p>";
+        $success = false;
+    }
+
+    return $success;
+}
+
+// Split a string of form "user@example.com" or "Example User <user@example.com>"
+// into the name and the email address.
+// Name will be null if not specified.
+function get_name_address($string)
+{
+    $string = rtrim($string, '>');
+    $results = explode(' <', $string);
+    if (sizeof($results) === 1) {
+        array_unshift($results, null);
+    }
+    return $results;
+}
+
 function maybe_mail($to, $subject, $message, $additional_headers = null)
```

Comment:
I'd love for us to think if there's a way we can make `maybe_mail()` end up being a wrapper around the new function so we only have one function that actually sends email.

It looks like there are only two places where we use `$additional_headers` for something besides `$from` and that's now supported by your new function call: `notify_old_pp.php` and `access_change_emails.inc`:
```php
    $headers = ["Reply-To: $db_requests_email_addr"];
    $mail_accepted = maybe_mail($email, $subject, $message_string, $headers);
```

It would mean that the new function would need to know the format of `$message` or we have a different way of enabling that via a couple of different functions.

Reply:
The new function supports `$from` and `$reply-to` - is there anything else that's needed?
Instead of making maybe_mail a wrapper, might it be easier to just fix the text of all the emails so they can all go through the new function - it didn't seem like there are too many to deal with. Then maybe_mail would be deleted, i.e. instead of having an intermediate version where both functions exist, just bring forward the final aim of a single function by fixing all the calls?

Reply:
I'm on-board with doing a wholesale update. I see one use of the old function in noncvs scripts, but that's an easy replacement.